sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/implementation/TableAccountSasGenerator.java

```diff
@@ -0,0 +1,114 @@
+import com.azure.data.tables.sas.TableAccountSasPermission;
+import com.azure.data.tables.sas.TableAccountSasSignatureValues;
+import com.azure.data.tables.sas.TableSasIpRange;
+import com.azure.data.tables.sas.TableSasProtocol;
+
+import java.time.OffsetDateTime;
+import java.util.Objects;
+
+import static com.azure.data.tables.implementation.TableSasUtils.computeHMac256;
+import static com.azure.data.tables.implementation.TableSasUtils.formatQueryParameterDate;
+import static com.azure.data.tables.implementation.TableSasUtils.logStringToSign;
+import static com.azure.data.tables.implementation.TableSasUtils.tryAppendQueryParameter;
+
+/**
+ * A class containing utility methods for generating SAS tokens for the Azure Storage accounts.
+ */
+public class TableAccountSasGenerator {
+    private final ClientLogger logger = new ClientLogger(TableAccountSasGenerator.class);
+    private String version;
+    private TableSasProtocol protocol;
+    private OffsetDateTime startTime;
+    private OffsetDateTime expiryTime;
+    private String permissions;
+    private TableSasIpRange sasIpRange;
+    private String services;
+    private String resourceTypes;
+
+    /**
+     * Creates a new {@link TableAccountSasGenerator} with the specified parameters.
+     *
+     * @param sasValues The {@link TableAccountSasSignatureValues account signature values}.
+     */
+    public TableAccountSasGenerator(TableAccountSasSignatureValues sasValues) {
+        this.version = sasValues.getVersion();
+        this.protocol = sasValues.getProtocol();
+        this.startTime = sasValues.getStartTime();
+        this.expiryTime = sasValues.getExpiryTime();
+        this.permissions = sasValues.getPermissions();
+        this.sasIpRange = sasValues.getSasIpRange();
+        this.services = sasValues.getServices();
+        this.resourceTypes = sasValues.getResourceTypes();
+    }
+
+    /**
+     * Generates a Sas signed with a {@link AzureNamedKeyCredential}.
+     *
+     * @param azureNamedKeyCredential {@link AzureNamedKeyCredential}
+     * @param context Additional context that is passed through the code when generating a SAS.
+     * @return A String representing the Sas
+     */
+    public String generateSas(AzureNamedKeyCredential azureNamedKeyCredential, Context context) {
+        Objects.requireNonNull(azureNamedKeyCredential, "'azureNamedKeyCredential' cannot be null.");
+        Objects.requireNonNull(services, "'services' cannot be null.");
+        Objects.requireNonNull(resourceTypes, "'resourceTypes' cannot be null.");
+        Objects.requireNonNull(expiryTime, "'expiryTime' cannot be null.");
+        Objects.requireNonNull(permissions, "'permissions' cannot be null.");
+
+        if (CoreUtils.isNullOrEmpty(version)) {
+            version = StorageConstants.HeaderConstants.TARGET_STORAGE_VERSION;
+        }
+        String stringToSign = stringToSign(azureNamedKeyCredential);
+        logStringToSign(logger, stringToSign, context);
```

Comment:
Yes.

Reply:
We use `Context` to know if the `stringToSign` should be logged or not.

Reply:
I see that this is the pattern used in Storage but it's not really ideal to use `Context` for random behavior control. It was meant for being used in pipeline policies and generally required for service calls only. Alternatively, you could just include a getter method that returns `stringToSign` and give user the control to log it or not.
This will simplify the `generate*` method signature and also the user doesn't have to learn about random constants that they need to include in `Context` just to log something.

Reply:
Sounds good, we won't log the string by default and can provide a method for the user to obtain it and do whatever they want with it like you suggest.

Reply:
Actually, I don't think I'll expose a public `getStringToSign()` method for now. Let's wait and see if someone asks for it.